hpy/debug/src/_debugmod.c

```diff
@@ -15,6 +15,8 @@ HPyDef_METH(new_generation, "new_generation", new_generation_impl, HPyFunc_NOARG
 static UHPy new_generation_impl(HPyContext *uctx, UHPy self)
 {
     HPyContext *dctx = hpy_debug_get_ctx(uctx);
+    if (dctx == NULL)
```

Comment:
These checks were added after error report from infer. https://github.com/hpyproject/hpy/actions/runs/1968620957/jobs/2813414188

Reply:
Maybe we should really abort in `hpy_debug_ctx_get` instead of returning `HPy_NULL`:
```HPyContext * hpy_debug_get_ctx(HPyContext *uctx)
{
    HPyContext *dctx = &g_debug_ctx;
    if (uctx == dctx) {
        HPy_FatalError(uctx, "hpy_debug_get_ctx: expected an universal ctx, "
                             "got a debug ctx");
    }
    if (hpy_debug_ctx_init(dctx, uctx) < 0) {
        HPyErr_SetString(uctx, uctx->h_SystemError, "Could not create debug context");
        return NULL;
    }
```